common/namer/namer_test.go

```diff
@@ -0,0 +1,173 @@
+			tokens:    []string{"foo", "bar", "baz", "qux", "qux"},
+			noHash:    true,
+			expected:  "foo-ba-baz-qu-qux",
+		},
+		{
+			maxLength: 16,
+			tokens:    []string{"foo", "bar", "baz", "qux", "qux"},
+			noHash:    true,
+			expected:  "fo-bar-ba-qux-qu",
+		},
+		{
+			maxLength: 15,
+			tokens:    []string{"foo", "bar", "baz", "qux", "qux"},
+			noHash:    true,
+			expected:  "fo-ba-baz-qu-qu",
+		},
+		{
+			maxLength: 14,
+			tokens:    []string{"foo", "bar", "baz", "qux", "qux"},
+			noHash:    true,
+			expected:  "fo-ba-ba-qu-qu",
+		},
+		{
+			maxLength: 13,
+			tokens:    []string{"foo", "bar", "baz", "qux", "qux"},
+			noHash:    true,
+			expected:  "fo-ba-b-qu-qu",
+		},
+		{
+			maxLength: 12,
+			tokens:    []string{"foo", "bar", "baz", "qux", "qux"},
+			noHash:    true,
+			expected:  "fo-b-ba-q-qu",
+		},
+		{
+			maxLength: 11,
+			tokens:    []string{"foo", "bar", "baz", "qux", "qux"},
+			noHash:    true,
+			expected:  "f-ba-b-qu-q",
+		},
+		{
+			maxLength: 10,
+			tokens:    []string{"foo", "bar", "baz", "qux", "qux"},
+			noHash:    true,
+			expected:  "f-b-ba-q-q",
+		},
+		{
+			maxLength: 9,
+			tokens:    []string{"foo", "bar", "baz", "qux", "qux"},
+			noHash:    true,
+			expected:  "f-b-b-q-q",
+		},
+		// Some real cases with a stack name from the CI
+		// 37 is the maximum size of EKS node group names
+		{
+			maxLength: 37,
+			tokens:    []string{"ci-17317712-4670-eks-cluster", "linux", "ng"},
+			expected:  "ci-17317712-4670-eks-cluster-linux-ng", // No truncation needed
+		},
+		{
+			maxLength: 37,
+			tokens:    []string{"ci-17317712-4670-eks-cluster", "linux-arm", "ng"},
+			expected:  "ci-17317712-4670-eks-c-linux-a-ng-458",
+		},
+		{
+			maxLength: 37,
+			tokens:    []string{"ci-17317712-4670-eks-cluster", "bottlerocket", "ng"},
+			expected:  "ci-17317712-4670-eks--bottleroc-n-a2e",
+		},
+		// 32 is the maximum size of load-balancer names
+		{
+			maxLength: 32,
+			tokens:    []string{"ci-17317712-4670-eks-cluster", "fakeintake"},
+			expected:  "ci-17317712-4670-eks-fakeint-5f1",
+		},
+		{
+			maxLength: 32,
+			tokens:    []string{"ci-17317712-4670-eks-cluster", "nginx"},
+			expected:  "ci-17317712-4670-eks-cl-ngin-db3",
+		},
+		{
+			maxLength: 32,
+			tokens:    []string{"ci-17317712-4670-eks-cluster", "redis"},
+			expected:  "ci-17317712-4670-eks-cl-redi-7de",
+		},
+	} {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			require.Conditionf(t, func() bool { return outputOK(tt.maxLength, tt.tokens, tt.expected) }, "Expected output string %q doesn’t match expected properties", tt.expected)
+			noHash = tt.noHash
+			output := joinWithMaxLength(tt.maxLength, tt.tokens)
+			assert.Equal(t, tt.expected, output)
+		})
+	}
+}
+
+func outputOK(maxLength int, tokens []string, output string) bool {
+	full := strings.Join(tokens, "-")
+	if len(full) <= maxLength {
+		return output == full
+	}
+
+	return len(output) == maxLength
+}
+
+func FuzzJoinWithMaxLength(f *testing.F) {
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do look for something to sincerely praise. -->
**👏 praise**
😍